src/Symfony/Component/Env/Dotenv.php

```diff
@@ -0,0 +1,365 @@
+                        break;
+                    }
+                }
+                $value .= $this->data[$this->cursor];
+                ++$this->cursor;
+
+                if ($this->cursor === $this->end) {
+                    throw new FormatException('Missing quote to end the value', $this->createFormatExceptionContext());
+                }
+            }
+            if ("\n" === $this->data[$this->cursor]) {
+                throw new FormatException('Missing quote to end the value', $this->createFormatExceptionContext());
+            }
+            ++$this->cursor;
+            $this->skipComment();
+        } elseif ('"' === $this->data[$this->cursor]) {
+            ++$this->cursor;
+            while ('"' !== $this->data[$this->cursor] || ('\\' === $this->data[$this->cursor - 1] && '\\' !== $this->data[$this->cursor - 2])) {
+                $value .= $this->data[$this->cursor];
+                ++$this->cursor;
+
+                if ($this->cursor === $this->end) {
+                    throw new FormatException('Missing quote to end the value', $this->createFormatExceptionContext());
+                }
+            }
+            if ("\n" === $this->data[$this->cursor]) {
+                throw new FormatException('Missing quote to end the value', $this->createFormatExceptionContext());
+            }
+            ++$this->cursor;
+            $this->skipComment();
+            $value = str_replace(array('\\\\', '\\"', '\r', '\n'), array('\\', '"', "\r", "\n"), $value);
+        } else {
+            $notQuoted = true;
+            while ($this->cursor < $this->end && "\n" !== $this->data[$this->cursor] && !(' ' === $this->data[$this->cursor - 1] && '#' === $this->data[$this->cursor])) {
+                $value .= $this->data[$this->cursor];
+                ++$this->cursor;
+            }
+            $value = rtrim($value);
+            $this->skipComment();
+        }
+
+        $this->skipEmptyLines();
+
+        $currentValue = $value;
+        if (!$singleQuoted) {
+            $value = $this->resolveVariables($value);
+            $value = $this->resolveCommands($value);
+        }
+
+        if ($notQuoted && $currentValue == $value && preg_match('/\s+/', $value)) {
+            throw new FormatException('A value containing spaces must be surrounded by quotes', $this->createFormatExceptionContext());
+        }
+
+        return $value;
+    }
+
+    private function skipWhitespace()
+    {
+        $this->cursor += strspn($this->data, ' ', $this->cursor);
+    }
+
+    private function skipEmptyLines()
+    {
+        if (preg_match('/(\n+|^#[^\n]*(\n*|$))+/Asm', $this->data, $match, null, $this->cursor)) {
+            $this->moveCursor($match[0]);
+        }
+    }
+
+    private function skipComment()
+    {
+        if (preg_match('/ *#[^\n]*(\n*|$)/Asm', $this->data, $match, null, $this->cursor)) {
+            $this->moveCursor($match[0]);
+        }
+    }
+
+    private function resolveCommands($value)
```

Comment:
we should maybe disable commands on Windows, because escaping is hard there, and it might come later?